src/Elastic.Apm/DistributedTracing/TraceParent.cs

```diff
@@ -0,0 +1,177 @@
+		{
+			traceId = string.Empty;
+			prentId = string.Empty;
+			traceFields = 0;
+
+			var bestAttempt = false;
+
+			if (string.IsNullOrWhiteSpace(traceParentValue)) return false;
+
+			if (traceParentValue.Length < VersionPrefixIdLength || traceParentValue[VersionPrefixIdLength - 1] != '-') return false;
+
+			try
+			{
+				var versionArray = StringToByteArray(traceParentValue, 0, VersionLength);
+				if (versionArray[0] == 255)
+					return false;
+
+				if (versionArray[0] > 0)
+					// expected version is 00
+					// for higher versions - best attempt parsing of trace id, span id, etc.
+					bestAttempt = true;
+			}
+			catch
+			{
+				return false;
+			}
+
+			if (traceParentValue.Length < VersionAndTraceIdLength || traceParentValue[VersionAndTraceIdLength - 1] != '-') return false;
+
+			try
+			{
+				var traceIdVal =
+					traceParentValue.Substring(VersionPrefixIdLength,
+						TraceIdLength);
+
+				if (!IsHex(traceIdVal) || traceIdVal == "00000000000000000000000000000000")
+					return false;
+
+				traceId = traceIdVal;
+			}
+			catch (ArgumentOutOfRangeException)
+			{
+				return false;
+			}
+
+			if (traceParentValue.Length < VersionAndTraceIdAndSpanIdLength
+				|| traceParentValue[VersionAndTraceIdAndSpanIdLength - 1] != '-') return false;
+
+			try
+			{
+				var prentIdVal = traceParentValue.Substring(VersionAndTraceIdLength, SpanIdLength);
+				if (!IsHex(prentIdVal) || prentIdVal == "0000000000000000")
+					return false;
+
+				prentId = prentIdVal;
+			}
+			catch (ArgumentOutOfRangeException)
+			{
+				return false;
+			}
+
+			if (traceParentValue.Length < VersionAndTraceIdAndSpanIdLength + OptionsLength) return false;
+
+			try
+			{
+				var fields = StringToByteArray(traceParentValue, VersionAndTraceIdAndSpanIdLength, OptionsLength);
+
+				if (fields != null && fields.Length > 0)
+					traceFields = fields[0];
+			}
+			catch (ArgumentOutOfRangeException)
+			{
+				return false;
+			}
+
+			if (!bestAttempt && traceParentValue.Length != VersionAndTraceIdAndSpanIdLength + OptionsLength) return false;
+
+			// ReSharper disable once InvertIf - imo that would make it hard to read.
+			if (bestAttempt)
+			{
+				if (traceParentValue.Length > VersionAndTraceIdAndSpanIdLength + OptionsLength &&
+					traceParentValue[VersionAndTraceIdAndSpanIdLength + OptionsLength] != '-')
+					return false;
+			}
+
+			bool IsHex(IEnumerable<char> chars)
+			{
+				// ReSharper disable once LoopCanBeConvertedToQuery - I benchmarked, that'd make parsing ~2x slower, don't do it!
+				foreach (var c in chars)
+				{
+					var isHex = c >= '0' && c <= '9' ||
+						c >= 'a' && c <= 'f' ||
+						c >= 'A' && c <= 'F';
+
+					if (!isHex)
+						return false;
+				}
+				return true;
+			}
+
+			return true;
+		}
+
+		public static string BuildTraceparent(string traceId, string spanId)
+			=> $"00-{traceId}-{spanId}-01";
+
+		private static byte[] StringToByteArray(string src, int start = 0, int len = -1)
```

Comment:
Maybe `HexStringToByte...` would be a better name?

Reply:
Made it to `HexStringTwoCharToByte`.